File: src/main/java/com/prasad/nithin/xmas/entity/QuestionAnswer.java
package com.prasad.nithin.xmas.entity;

import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class QuestionAnswer {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	String id;

	@OneToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "question", referencedColumnName = "questionId")
	private Question question;

	@ManyToMany(fetch = FetchType.EAGER)
	@JoinColumn(name = "answer", referencedColumnName = "userId")
	
	Set<User> answer;
	
	TEAM team;

	private String trackingId;

}

File: src/main/ng-app/xmas-ui/src/app/datamodel/QuestionAnswer.ts
import { Question } from "./Question";
import { Teams } from "./Teams";
import { User } from "./User";

export class QuestionAnswer{
    constructor(private _id?:String,private _question?:Question,private _answer?:User[],private _team?:Teams,private _trackingId?:String){

    }


    get teams(){
        return this._team;
    }

    get question(){
        return this._question;
    }

    get answer():User[]{
        return this.answer;
    }

    getRequestObject() {
       let  userArr:Array<any>=[];
       this._answer?.forEach(data =>userArr.push({"userId":data.userId}));
       let data:any = {
            "question": {"questionId":this._question?.questionId},
            "answer":userArr ,
            "team": this._team
        }
        return data;
    }
}